pkgs/development/python-modules/nbqa/default.nix

```
@@ -0,0 +1,64 @@
+{ autopep8
+, black
+, buildPythonPackage
+, fetchPypi
+, flake8
+, ipython
+, isort
+, jupytext
+, lib
+, mypy
+, pylint
+, pythonAtLeast
+, pythonOlder
+, pyupgrade
+, ruff
+, tokenize-rt
+, tomli
+}:
+buildPythonPackage rec {
+  pname = "nbqa";
+  version = "1.7.0";
+
+  # TODO: Implement support for older Python 3.8 and 3.9, as supported by the project. Requires fixing jupyter-ydoc build.
+  disabled = pythonOlder "3.10" || pythonAtLeast "3.12";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-EXESrW1hj/E6/FukHKD/IcI/CEpt94dfsEt/LAGhNsQ=";
+  };
+
+  passthru.optional-dependencies = {
+    # TODO: Add blacken-docs; it's not yet packaged in nixpkgs.
+    all = [
+      black
+      flake8
+      isort
+      jupytext
+      mypy
+      pylint
+      pyupgrade
+      ruff
+    ];
+    black = [ black ];
+    flake8 = [ flake8 ];
+    isort = [ isort ];
+    jupytext = [ jupytext ];
+    mypy = [ mypy ];
+    pylint = [ pylint ];
+    pyupgrade = [ pyupgrade ];
+    ruff = [ ruff ];
+  };
+
+  propagatedBuildInputs = [ autopep8 ipython tokenize-rt tomli ] ++ passthru.optional-dependencies.all;
```

```suggestion
  propagatedBuildInputs = [ autopep8 ipython tokenize-rt tomli ] ++ passthru.optional-dependencies.all;
  
  nativeBuildInputs = [
    pytestCheckHook
  ];
```
Would you please add `pytestCheckHook` for tests?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

---

I already have `nativeCheckInputs = [ pytestCheckHook ];`, and the tests run during the build. Does adding it to `nativeBuildInputs` change anything?

---

Sorry about the nativeBuildInputs, my mistake.
But nativeCheckInputs is not defined anywhere.
Have you read the document I suggested as a reference?

---

>  But nativeCheckInputs is not defined anywhere.

You're absolutely right, my mistake. The PyPI package does not include tests, does that mean I should fetch from GitHub instead?

---

Yes, please do so.

---

Done. I'll need some time/help to get the tests to pass, however.

---

Most of the tests pass now. Not sure if that's good enough, but I'm not sure why most of them are failing. It seems to be that they can't find test files, which might indicate that the working directory is not the repository.